check_codegen.sh

```diff
@@ -1,12 +1,29 @@
 # This is a script which checks if yarn codegen was run it used by the check_codegen evergreen task
-BEFORE=$(git status --porcelain | grep src/gql/generated/types.ts)
-yarn codegen
-AFTER=$(git status --porcelain | grep src/gql/generated/types.ts)
+
+# path to generated types
+types_file="src/gql/generated/types.ts"
+{
+  BEFORE=$(git status --porcelain | grep "$types_file")
+  yarn codegen
+  AFTER=$(git status --porcelain | grep "$types_file")
+} || {
+  echo "script failed"
+  exit 1
+}
+
 if [ "$BEFORE" != "$AFTER" ]; then
-  echo "src/gql/types.ts is not up to date"
+  echo "$types_file is not up to date"
   echo "did you forget to run 'yarn codegen'?"
+  # Check if bin directory exists so we can save the diff to it
+  if [ -d "bin" ] 
+  then
+      echo "Directory /bin already exists." 
+  else
+      mkdir bin
+  fi
+  git diff $types_file >> bin/codegen.diff
   exit 1
 else
-  echo "src/gql/types.ts is up to date"
-  exit 0
```

Comment:
The script was always exiting 0 despite the `yarn codegen` task exiting 1 which would make the task misleadingly succeed. I removed that default exit 0 condition and added a guard above to exit if the script ever fails. cc @minnakt @sophstad